src/main/kotlin/org/jetbrains/research/testspark/helpers/psiHelpers/JavaPsiHelper.kt

```diff
@@ -173,145 +136,313 @@ class JavaPsiHelper : PsiHelperInterface {
+                            JavaPsiClassWrapper(typeIt).let {
+                                if (!it.qualifiedName.startsWith("java.")) {
+                                    interestingPsiClasses.add(it)
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+            currentLevelClasses = mutableListOf<PsiClassWrapper>().apply { addAll(tempListOfClasses) }
+            interestingPsiClasses.addAll(tempListOfClasses)
+        }
+        log.info("There are ${interestingPsiClasses.size} interesting psi classes")
+        return interestingPsiClasses.toMutableSet()
+    }
+
+    override fun getInterestingPsiClassesWithQualifiedNames(
+        cut: PsiClassWrapper,
+        psiMethod: PsiMethodWrapper,
+    ): MutableSet<PsiClassWrapper> {
+        val interestingPsiClasses = cut.getInterestingPsiClassesWithQualifiedNames(psiMethod)
+        log.info("There are ${interestingPsiClasses.size} interesting psi classes from method ${psiMethod.methodDescriptor}")
+        return interestingPsiClasses
+    }
+
+    override fun getCurrentListOfCodeTypes(e: AnActionEvent): Array<*>? {
+        val result: ArrayList<String> = arrayListOf()
+        val caret: Caret =
+            e.dataContext.getData(CommonDataKeys.CARET)?.caretModel?.primaryCaret ?: return result.toArray()
+
+        val javaPsiClassWrapped = getSurroundingClass(caret.offset) as JavaPsiClassWrapper?
+        val javaPsiMethodWrapped = getSurroundingMethod(caret.offset) as JavaPsiMethodWrapper?
+        val line: Int? = getSurroundingLine(caret.offset)?.plus(1)
+
+        javaPsiClassWrapped?.let { result.add(getClassDisplayName(it)) }
+        javaPsiMethodWrapped?.let { result.add(getMethodDisplayName(it)) }
+        line?.let { result.add(getLineDisplayName(it)) }
+
+        if (javaPsiClassWrapped != null && javaPsiMethodWrapped != null) {
+            log.info(
+                "The test can be generated for: \n " +
+                    " 1) Class ${javaPsiClassWrapped.qualifiedName} \n" +
+                    " 2) Method ${javaPsiMethodWrapped.methodDescriptor}" +
+                    " 3) Line $line",
+            )
+        }
+
+        return result.toArray()
+    }
+
+    override fun getLineDisplayName(line: Int): String {
+        return "<html><b><font color='orange'>line</font> $line</b></html>"
+    }
+
+    override fun getClassDisplayName(psiClass: PsiClassWrapper): String {
+        return if ((psiClass as JavaPsiClassWrapper).isInterface) {
             "<html><b><font color='orange'>interface</font> ${psiClass.qualifiedName}</b></html>"
-        } else if (isAbstractClass(psiClass)) {
+        } else if (psiClass.isAbstractClass) {
             "<html><b><font color='orange'>abstract class</font> ${psiClass.qualifiedName}</b></html>"
         } else {
             "<html><b><font color='orange'>class</font> ${psiClass.qualifiedName}</b></html>"
         }
```

Comment:
The same method is implemented by Arkadii in the PR I mentioned before.
@arksap2002

Reply:
I canceled that PR due to incorrect implementation in case of language increasing

Reply:
pushed a new changes in this branch